### PR TITLE
New tests and updated readme

### PR DESCRIPTION
**Pull request by beccaelenzil**
The tests have been updated to mostly use data in `tests_constants.py.`
- The user_data for each wave builds on the user_data from previous waves so that the data structure is consistent throughout.
- The full test suite is provided here. Once we are confident this is test suite is complete, I will remove some of the steps from the tests (Waves 02 and later) to scaffold students learning to write tests.

The README has been updated to emphasize TDD
- As part of the workflow, students need to check if a test is incomplete, and if it is, complete it
- I made the decision to remove the extra section on submission/workflow because it seemed to repeat the workflow above, but simplify it in a way that leaves out steps.
- A `play_tester.py` file is provided and there are suggestions on how to use it

This fork/branch has a complete solution with this test suite: https://github.com/beccaelenzil/viewing-party/tree/be/new-tests

### CODE REVIEW

**Review by jbieniosek — APPROVED** [review]

File: tests/test_wave_03.py [hunk 1]
```diff
@@ -1,212 +1,77 @@
...
-                        "title": "Title D"
-                    },
-                    {
-                        "title": "Title E"
-                    }
-                ]
-            }
-        ]
-    }
+    amandas_data = USER_DATA_3
```
[the remark this answers is not in the document]
Comment (goeunpark) [reply tests/test_wave_03.py:34]: something something pytest fixtures instead of using mutable global variables? (sry, still a a python n00b ;__; )

File: README.md [hunk 2]
```diff
@@ -141,19 +156,26 @@ $ pytest tests/test_file_name.py
...
-1. Review your code in `main.py` and see if there are ways you can make the code more readable.
-1. Then, work on making the tests in `test_wave_02.py` pass.
-1. Review your code in `main.py`
-1. Repeat on all test files until submission time.
+```bash
+python3 play_tester.py
+```
 
-At submission time, no matter where you are, submit the project via Learn.
+There is some starter code provided in `play_tester.py`. This code prints the test data that is used for many of the tests. Looking closely at this data can help us think critically about the expected output for given input for each function. Then, calling each function with this data allows us to observe the **actual** output for given input.
```
Comment (goeunpark) [inline README.md:178]: this is so helpful, esp since we cleaned up our test files and can no longer just check the shape of the input by glancing at the test

File: tests/test_wave_03.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (goeunpark) [reply tests/test_wave_03.py:34]: update: this is likely happening because my implementation of `get_unique_watched` mutates the `user_data` 😬

[the remark this answers is not in the document]
Comment (beccaelenzil) [reply tests/test_wave_03.py:34]: per our conversation, we don't want students to mutate `user_data`, but it's also a problem that our unit tests don't start fresh data each time. I will think more on how to handle this. Introducing fixtures now feels too early, so it may mean copying and pasting the data into each test.

[the remark this answers is not in the document]
Comment (beccaelenzil) [reply tests/test_wave_03.py:34]: The approach of pulling the data out into a `test_constants.py` file came from writing integration tests. We decided to scrap the integration tests because they were really unit tests stuck together, but I think this same approach doesn't work well for unit tests.

File: tests/test_wave_01.py [hunk 3]
```diff
@@ -154,33 +142,17 @@ def test_moves_movie_from_watchlist_to_watched():
...
-                "rating": 3.9
-            }
-        ]
+        "watchlist": [FANTASY_1],
+        "watched": [FANTASY_2]
     }
 
     # Act
-    updated_data = watch_movie(janes_data, "Title A")
+    updated_data = watch_movie(janes_data, "Non-Existent Movie Title")
```
Comment (kaidamasaki) [inline tests/test_wave_01.py:155]: I really like this clarification in the test.

**Review by kaidamasaki — COMMENTED** [review]
Looks pretty good.  Just a few suggestions, mostly around not using `is` and making deep copies.

File: README.md [hunk 4]
```diff
@@ -85,54 +87,67 @@ $ source venv/bin/activate
 
 2. Find the test file that contains the test you want to run. Ensure that the test(s) you want to run isn't skipped.
 
-   - Check the `tests` folder, and find the test file you want to run
+    - Check the `tests` folder, and find the test file you want to run
```
[the remark this answers is not in the document]
Comment (beccaelenzil) [reply README.md:90]: Actually no! It reads a little bit more clearly with the double indent

File: README.md [hunk 5]
```diff
@@ -85,54 +87,67 @@ $ source venv/bin/activate
...
    - In that test file, read through each test case
-   - Remove all lines that contain `@pytest.mark.skip()`
+      - If it is incomplete, complete the test. 
+        - *Is this a nominal or edge case?*
```
[the remark this answers is not in the document]
Comment (beccaelenzil) [reply README.md:93]: Which lesson? Intro to Automated Tests or Intro to Pytest

[the remark this answers is not in the document]
Comment (beccaelenzil) [reply README.md:93]: Can't link to lessons from repos since students are in different classes.

**Comment by beccaelenzil** [comment]
Based on feedback, the following changes were made:
* Replaced is with ==
* Implemented functions to provide a deep copy of USER_DATA for each test to mimic fixtures
* Updated Readme to describe test_constants
* Update Readme to add note about not modifying the data

**Comment by beccaelenzil** [comment]
Here is a separate PR removing act and/or assert for 4 tests (Waves 01, 03, and 05): https://github.com/beccaelenzil/viewing-party/pull/2/files

File: README.md [hunk 6]
```diff
@@ -141,19 +158,32 @@ $ pytest tests/test_file_name.py
...
-1. Review your code in `main.py`
-1. Repeat on all test files until submission time.
+While we will mainly use a Test Driven Development (TDD) workflow for this project, it can be helpful to run code independently from running tests. To do this, a file `play_tester.py` is provided.
 
-At submission time, no matter where you are, submit the project via Learn.
+To run this file, use:
+
+```bash
+python3 play_tester.py
+```
```
Comment (goeunpark) [inline README.md:178]: This also has to be run in venv, might be worth adding `# Must be in activated virtual environment` NBD if it's overkill though

**Comment by beccaelenzil** [comment]
Merged with this PR: https://github.com/AdaGold/viewing-party/pull/18
